trieste/acquisition/function/function.py

```diff
@@ -1034,6 +1035,595 @@ def __call__(self, x: TensorType) -> TensorType:
         return tf.reduce_mean(batch_improvement, axis=-1, keepdims=True)  # [..., 1]
 
 
+class BatchExpectedImprovement(SingleModelAcquisitionBuilder[ProbabilisticModel]):
+    """Accurate approximation of the batch expected improvement, using the
+    method of Chvallier and Ginsbourger :cite:`chevalier2013fast`.
+
+    Internally, this uses a highly accurate approximation of the cumulative
+    density function of the multivariate Gaussian, developed by Alan Genz
+    :cite:`genz2016numerical`.
+    """
+
+    def __init__(
+        self,
+        sample_size: int,
+        *,
+        batch_size: int,
```

Comment:
I don't like that we need to specify batch size here. We dont do this for any of our other batch acquisition functions. They should work out the batch from the size of the input and recalc stuff or crash if the batch size changes. See the MonteCarloBatchEI code. @uri-granta please can you assist with this

Reply:
Yes, this is a good idea. Changed this to infer the size of the batch at runtime and throw an error if it changes.